plugins/web/src/server_command.cpp

```diff
@@ -391,56 +390,60 @@ auto server_command(const vast::invocation& inv, caf::actor_system& system)
-      using request_handler_t = restinio::router::express_router_t<>;
-    };
-    VAST_INFO("web plugin listening on http://{}:{}", config.bind_address,
-              config.port);
-    restinio::run(restinio::on_this_thread<my_server_traits>()
-                    .port(config.port)
-                    .address(config.bind_address)
-                    .request_handler(std::move(router)));
-  } else {
-    using traits_t = restinio::single_thread_tls_traits_t<
-      restinio::asio_timer_manager_t, restinio::single_threaded_ostream_logger_t,
-      restinio::router::express_router_t<>>;
-
-    asio::ssl::context tls_context{asio::ssl::context::tls};
-    // Most examples also set `asio::ssl::context::default_workarounds`, but
-    // based on [1] these are only relevant for SSL which we don't support
-    // anyways.
-    // [1]: https://www.openssl.org/docs/man1.0.2/man3/SSL_CTX_set_options.html
-    tls_context.set_options(asio::ssl::context::tls_server
-                            | asio::ssl::context::single_dh_use);
-    if (server_config->require_clientcerts)
-      tls_context.set_verify_mode(
-        asio::ssl::context::verify_peer
-        | asio::ssl::context::verify_fail_if_no_peer_cert);
-    else
-      tls_context.set_verify_mode(asio::ssl::context::verify_none);
-    tls_context.use_certificate_chain_file(config.certfile);
-    tls_context.use_private_key_file(config.keyfile, asio::ssl::context::pem);
-    // Manually specifying DH parameters is deprecated in favor of using
-    // the OpenSSL built-in defaults, but asio has not been updated to
-    // expose this API so we need to use the raw context.
-    SSL_CTX_set_dh_auto(tls_context.native_handle(), true);
-    VAST_INFO("web plugin listening on https://{}:{}", config.bind_address,
-              config.port);
-    using namespace std::literals::chrono_literals;
-    restinio::run(restinio::on_this_thread<traits_t>()
-                    .address(config.bind_address)
-                    .port(config.port)
-                    .request_handler(std::move(router))
-                    .read_next_http_message_timelimit(10s)
-                    .write_http_response_timelimit(1s)
-                    .handle_request_timeout(1s)
-                    .tls_context(std::move(tls_context)));
-  }
-  self->send_exit(dispatcher, caf::error{});
+  auto io_context = asio::io_context{};
+  auto server = make_server(*server_config, std::move(router),
+                            restinio::external_io_context(io_context));
+  // Post initial action to asio event loop. Note that the action
+  // must have been posted *before* calling `io_context.run()`.
+  asio::post(io_context, [&] {
+    std::visit(detail::overload{[](auto& server) {
+                 server->open_sync();
+               }},
+               server);
+  });
+  // Launch the thread on which the server will work.
+  std::thread server_thread{[&] {
```

Comment:
A word of warning; I had a bit of a headache concerning the lifetimes of variables when capturing local variables by reference here but ultimatately I thought that since we're blocking below until the thread is joined again this *should* be safe.